### PR TITLE
batch send raft log gc command

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -1196,13 +1199,16 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         self.region_ranges.insert(enc_end_key(&region), region.get_id());
     }
 
-    fn on_ready_result(&mut self, region_id: u64, exec_results: Vec<ExecResult>) {
+    fn on_ready_result(&mut self,
+                       region_id: u64,
+                       exec_results: Vec<ExecResult>,
+                       raftlog_gc_tasks: &mut RaftlogGcTasks) {
         // handle executing committed log results
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: I think we can handle task here directly, no need pass the arg and handle outside.

File: src/raftstore/store/store.rs [hunk 2]
```diff
@@ -535,13 +535,19 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         loop {
             match self.apply_res_receiver.as_ref().unwrap().try_recv() {
                 Ok(ApplyTaskRes::Applys(multi_res)) => {
+                    let mut raftlog_gc_tasks = RaftlogGcTasks::from_vec(vec![]);
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: `from_vec(vec![])` seems strange

File: src/raftstore/store/store.rs (hunk 1 above)
Comment (zhangjinpeng1987) [inline src/raftstore/store/store.rs]: One region contains only one compact log result at most.

Comment (siddontang) [inline src/raftstore/store/store.rs]: oh, my fault.

File: src/raftstore/store/store.rs [hunk 3]
```diff
@@ -535,13 +535,19 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         loop {
             match self.apply_res_receiver.as_ref().unwrap().try_recv() {
                 Ok(ApplyTaskRes::Applys(multi_res)) => {
+                    let mut raftlog_gc_tasks = RaftlogGcTasks::new();
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: can we use` len(multi_res)` as the initialized capacity? In most cases, the ready res has only CompactLog command.

**Pull request closed without merging**